src/screens/barcodeScan/BarcodeScan.js

```diff
@@ -1,205 +1,426 @@
+							cutElement.substring(cutElement).indexOf("DL") + 2; // find second instance of DL in element
+						elements[
+							cutElement.substring(indexOfData, indexOfData + 3)
+						] = cutElement.substring(indexOfData + 3);
+					}
+					if (element[0] == "D") {
+						elements[element.substring(0, 3)] =
+							element.substring(3);
+					}
+					c;
+				}
+				const id = this.props.navigation.state.params.objectID;
+				if ("DAC" in elements) {
+					// add first name
+					this.props.updateDriver({
+						id,
+						question: Constants.FIRST_NAME_ID,
+						selection: elements["DAC"],
+					});
+				}
+				if ("DAD" in elements) {
+					// add middle inital
+					this.props.updateDriver({
+						id,
+						question: Constants.MIDDLE_INITIAL_ID,
+						selection: elements["DAD"][0],
+					});
+				}
+				if ("DCS" in elements) {
+					// add last name
+					this.props.updateDriver({
+						id,
+						question: Constants.LAST_NAME_ID,
+						selection: elements["DCS"],
+					});
+				}
+				if ("DBB" in elements) {
+					// add date of birth
+					const dob = elements["DBB"];
+					const formattedDOB =
+						dob.substring(dob.length - 4) +
+						"/" +
+						dob.substring(0, 2) +
+						"/" +
+						dob.substring(2, 4);
+					this.props.updateDriver({
+						id,
+						question: Constants.DATE_OF_BIRTH_ID,
+						selection: formattedDOB,
+					});
+				}
+				if ("DAG" in elements) {
+					// add street address
+					this.props.updateDriver({
+						id,
+						question: Constants.STREET_ADDRESS_ID,
+						selection: elements["DAG"],
+					});
+				}
+				if ("DAI" in elements) {
+					// add city
+					this.props.updateDriver({
+						id,
+						question: Constants.CITY_ID,
+						selection: elements["DAI"],
+					});
+				}
+				if ("DAJ" in elements) {
+					// add state
+					this.props.updateDriver({
+						id,
+						question: Constants.STATE_ID,
+						selection: elements["DAJ"],
+					});
+				}
+				if ("DAK" in elements) {
+					// add ZIP code
+					var zip = elements["DAK"];
+					if (zip.substring(zip.length - 4) == "0000") {
+						// if last four digits are unknown, ignore them
+						zip = zip.substring(0, zip.length - 4);
+					}
+					this.props.updateDriver({
+						id,
+						question: Constants.ZIP_CODE_ID,
+						selection: zip,
+					});
+				}
+				if ("DBC" in elements) {
+					// add sex
+					var sex = elements["DBC"];
+					sex = sex == 2 ? 0 : sex; // if sex is female(2), change to correct id (0)
+					this.props.updateDriver({
+						id,
+						question: Constants.SEX_ID,
+						selection: sex,
+					});
+				}
+				if ("DAQ" in elements) {
+					// add driver's license id
+					this.props.updateDriver({
+						id,
+						question: Constants.DLICENSE_ID,
+						selection: elements["DAQ"],
+					});
+				}
+				if ("DCD" in elements) {
+					// add license endorsements
+					const endorsmentValue = elements["DCD"].toUpperCase();
+					var endorsementCode = "";
+					switch (endorsmentValue) {
+						case "NONE":
```

Comment:
could we turn this switch statement into a dictionary, where the keys are the cases and the values are the corresponding endorsement codes? and then we don't need to repeat so much code